src/server/api/routes/users/v2/email.js

```diff
@@ -31,14 +31,16 @@ router.post('/email', authenticateV2Login, async (req, res, next) => {
     const token = uuidv4();
     await db.runTransaction(async t => {
       const userRef = walletUserCollection.doc(user);
-      const userDoc = await t.get(userRef);
-      const {
-        email: currentEmail,
-        emailUnconfirmed,
-        emailLastUpdatedTs,
-      } = userDoc.data();
-      if (email === currentEmail) {
-        throw new Error('EMAIL_ALREADY_UPDATED');
+      const [userDoc, userDocWithSameEmail] = await Promise.all([
+        t.get(userRef),
+        t.get(walletUserCollection.where('email', '==', email).limit(1)),
+      ]);
+      const { emailUnconfirmed, emailLastUpdatedTs } = userDoc.data();
```

Comment:
Seems this line can be put after the if block below

Reply:
Yes, that would be better!